src/xcode/ENA/ENA/Source/Services/Risk/Provider/RiskProvider.swift

```diff
@@ -434,15 +434,17 @@ final class RiskProvider: RiskProviding {
 		store.enfRiskCalculationResult = enfRiskCalculationResult
 		store.checkinRiskCalculationResult = checkinRiskCalculationResult
 
+		// first check if a notification might get triggered then write mostRecentDateWithRiskLevel to the store
 		checkIfRiskLevelHasChangedForNotifications(risk)
+		store.mostRecentDateWithRiskLevel = risk.details.mostRecentDateWithRiskLevel
```

Comment:
Maybe it would be better to save the date inside of `checkIfRiskLevelHasChangedForNotifications`.
This way its clearer, that it belongs to the algorithm of `checkIfRiskLevelHasChangedForNotifications` and its unlikely that the order will be changed here.